routes/users.js

```diff
@@ -135,4 +134,35 @@ router.get("/:id", async (req, res, next) => {
   }
 });
 
+router.patch("/:id", async (req, res, next) => {
+  const { nickname, imageUrl } = req.body;
+  const { auth: token } = req.cookies;
+
+  const { _id: userId } = jwt.decode(token);
+
+  try {
+    if (nickname === undefined && imageUrl === undefined) {
+      throw createError(422, INVALID_REQUEST);
+    }
+
+    if (nickname && imageUrl) {
+      await User.findByIdAndUpdate(userId, { nickname, imageUrl });
+    } else if (nickname) {
+      await User.findByIdAndUpdate(userId, { nickname });
+    } else if (imageUrl) {
+      await User.findByIdAndUpdate(userId, { imageUrl });
+    }
```

Comment:
```suggestion
    const updateData = {};
    
    nickname && (updateData.nickname = nickname);
    imageUrl && (updateData.imageUrl = imageUrl);
    
    await User.findByIdAndUpdate(userId, updateData);
```

반복되는 로직을 이런식으로도 표현할 수 있습니다~!

Reply:
간결해서 파악하기 한결 수월해지네요! 수정하겠습니다!